test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -246,115 +253,123 @@ TEST_F(FakePeerExampleFixture, SynchronizeTheRightVersionOfForkedLedger) {
   valid_block_storage->storeBlock(new_valid_block);
 
   // wait for the real peer to commit the blocks and check they are from the
-  // vlid branch
-  std::mutex wait_mutex;
-  std::condition_variable wait_cv;
-  std::atomic_bool commited_all_blocks(false);
+  // valid branch
   itf.getIrohaInstance()
       .getIrohaInstance()
       ->getStorage()
       ->on_commit()
-      .subscribe([&wait_cv, &commited_all_blocks, &valid_block_storage](
-                     const auto &committed_block) {
+      .tap([&valid_block_storage](
+               const std::shared_ptr<shared_model::interface::Block>
+                   &committed_block) {
         const auto valid_hash =
             valid_block_storage->getBlockByHeight(committed_block->height())
                 ->hash()
                 .hex();
         const auto commited_hash = committed_block->hash().hex();
-        if (commited_hash != valid_hash) {
+        if (commited_hash == valid_hash) {
           ASSERT_EQ(commited_hash, valid_hash) << "Wrong block got committed!";
```

Comment:
Looks like conditional in previous line should be removed, otherwise `ASSERT_EQ` will always pass.